src/test/java/org/elasticsearch/action/admin/cluster/snapshots/status/SnapshotStatusTests.java

```diff
@@ -0,0 +1,44 @@
+/*
+ * Licensed to Elasticsearch under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.elasticsearch.action.admin.cluster.snapshots.status;
+
+import org.elasticsearch.action.admin.cluster.snapshots.SnapshotTests;
+import org.junit.Test;
+
+import static org.hamcrest.Matchers.equalTo;
+import static org.hamcrest.Matchers.hasSize;
+
+public class SnapshotStatusTests extends SnapshotTests {
+
+    @Test
+    public void testSnapshotStatusWithBlocks() {
+        // This test checks that the Snapshot Status operation is never blocked, even if the cluster is read only.
+        try {
+            setClusterReadOnly(true);
+            SnapshotsStatusResponse response = client().admin().cluster().prepareSnapshotStatus(REPOSITORY_NAME)
+                                                                            .setSnapshots(SNAPSHOT_NAME)
+                                                                            .execute().actionGet();
+            assertThat(response.getSnapshots(), hasSize(1));
+            assertThat(response.getSnapshots().get(0).getState().completed(), equalTo(true));
+        } finally {
+            setClusterReadOnly(false);
+        }
+    }
+}
```

Comment:
here too I wonder if we should merge all these snashot tests in the same class, since each one of them have a single test method.